### PR TITLE
Add method to find widget by data ui attribute

### PR DESCRIPTION
**Pull request by khaydushki**
Changes made:
- Added dataUiSelector in the FindElementOptions (Find the widget by data ui attribute)

Signed-off-by: Krum Haydushki <khaydushki@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Examples have been added / updated (for bug fixes / features)
-   [x] Changelog has been updated

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Example website changes
-   [ ] Version bump
-   [ ] Other... Please describe:

## What does this change do?
Add method to find widget by data ui attribute.

## What manual testing did you do?
Ran the unit tests and observe them passing

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/components/src/utils/test/widget-object/angular/angular-widget-finder.ts [hunk 1]
```diff
@@ -57,8 +58,8 @@ export class AngularWidgetObjectFinder<H = unknown> {
             new TestElement([ancestor ? ancestor : this.fixture.debugElement], this.fixture)
         );
         let query = widgetConstructor.tagName;
-        if (findOptions?.cssSelector) {
-            query = query + findOptions.cssSelector;
+        if (findOptions) {
+            query = query + SelectorUtil.extractSelector(findOptions);
```
Comment (ryan-bradford) [inline projects/components/src/utils/test/widget-object/angular/angular-widget-finder.ts:62]: This query is passed to `get` as `cssSelector`. So including the data-ui here will make it duplicated, because you are also adding it in a the call to get.

Undo this change, and instead add the data-ui in `parentQuery` below. And please make a similar change in the cypress-widget-finder.

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts [hunk 2]
```diff
@@ -37,8 +38,8 @@ export class CypressWidgetObjectFinder<T> {
         findOptions?: FindCypressWidgetOptions
     ): W {
         let tagName = widgetConstructor.tagName;
-        if (findOptions?.cssSelector) {
-            tagName += `${findOptions.cssSelector}`;
+        if (findOptions) {
+            tagName += SelectorUtil.extractSelector(findOptions);
```
Comment (ryan-bradford) [inline projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:42]: I should have deleted these 4 lines of code. You should be able to remove them. `let tagName` is not used.

**Review by ryan-bradford — CHANGES_REQUESTED** [review]

File: projects/components/src/utils/test/widget-object/angular/angular-widget-finder.ts (hunk 1 above)
Comment (khaydushki) [reply projects/components/src/utils/test/widget-object/angular/angular-widget-finder.ts]: Done for both widget-finders.

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts (hunk 2 above)
Comment (khaydushki) [reply projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts]: I removed them.

**Review by ryan-bradford — APPROVED** [review]
Looks great. As we discussed, just make it so only either `dataUiSelector` or `cssSelector` applies.
